src/QSymbolicsBase/rules.jl

```diff
@@ -40,7 +47,83 @@ end
 
 FLATTEN_RULES = [
     @rule(~x::isnotflat_precheck(⊗) => flatten_term(⊗, ~x)),
-    @rule ⊗(~~xs::hasscalings) => prefactorscalings_rule(xs)
+    @rule ⊗(~~xs::hasscalings) => prefactorscalings_rule(xs) # Used to perform (a*|k⟩) ⊗ (b*|l⟩) → (a*b) * (|k⟩⊗|l⟩) 
 ]
 
 tensor_simplify = Fixpoint(Chain(FLATTEN_RULES))
+
+"""Quantum circuit identities"""
+_isXX(x) = x isa SApplyOp && x.terms == [X,X] # work on this with identity op
+_isYY(x) = x isa SApplyOp && x.terms == [Y,Y]
+_isZZ(x) = x isa SApplyOp && x.terms == [Z,Z]
+_isXY(x) = x isa SApplyOp && x.terms == [X,Y]
+_isYZ(x) = x isa SApplyOp && x.terms == [Y,Z]
+_isZX(x) = x isa SApplyOp && x.terms == [Z,X]
+_isYX(x) = x isa SApplyOp && x.terms == [Y,X]
+_isZY(x) = x isa SApplyOp && x.terms == [Z,Y]
+_isXZ(x) = x isa SApplyOp && x.terms == [X,Z]
+_isHXH(x) = x isa SApplyOp && x.terms == [H,X,H]
+_isHYH(x) = x isa SApplyOp && x.terms == [H,Y,H]
+_isHZH(x) = x isa SApplyOp && x.terms == [H,Z,H]
+
+
+CIRCUIT_RULES = [
+    @rule(~x::_isXX => I),
+    @rule(~x::_isYY => I),
+    @rule(~x::_isZZ => I),
+    @rule(~x::_isXY => im*Z),
+    @rule(~x::_isYZ => im*X),
+    @rule(~x::_isZX => im*Y),
+    @rule(~x::_isYX => -im*Z),
+    @rule(~x::_isZY => -im*X),
+    @rule(~x::_isXZ => -im*Y),
+    @rule(~x::_isHXH => Z),
+    @rule(~x::_isHYH => -Y),
+    @rule(~x::_isHZH => X)
+]
+
+circuit_simplify = Fixpoint(Chain(CIRCUIT_RULES))
+
+"""Commutator identities"""
+_isXYcommutator(x) = x isa SCommutator && _isX(x.op1) && _isY(x.op2)
+_isYZcommutator(x) = x isa SCommutator && _isY(x.op1) && _isZ(x.op2)
+_isZXcommutator(x) = x isa SCommutator && _isZ(x.op1) && _isX(x.op2)
+_isYXcommutator(x) = x isa SCommutator && _isY(x.op1) && _isX(x.op2)
+_isZYcommutator(x) = x isa SCommutator && _isZ(x.op1) && _isY(x.op2)
+_isXZcommutator(x) = x isa SCommutator && _isX(x.op1) && _isZ(x.op2)
+
+COMMUTATOR_RULES = [
+    @rule(~x::_isXYcommutator => 2*im*Z),
```

Comment:
Same comment: this should use pattern matching, not such complicated rules.

Reply:
given my other comment, this should be as simple as `@rule commutator(~o1::_isa(XGate), ~o2::_isa(Ygate)) => 2im Z`